### PR TITLE
fix incorrect split for select query

### PR DESCRIPTION
**Pull request by LYHuang**
- add new split function for brackets and quotes selection 
- should only affect SELECT section in search component
- add test

image

Ref: hdx-1587

### CODE REVIEW

File: packages/app/src/components/DBRowTable.tsx [hunk 1]
```diff
@@ -699,7 +702,7 @@ function mergeSelectWithPrimaryAndPartitionKey(
     primaryKeys.trim() !== '' ? splitAndTrimCSV(primaryKeys) : [];
   const allKeys = [...partitionKeyArr, ...primaryKeyArr];
   if (typeof select === 'string') {
-    const selectSplit = splitAndTrimCSV(select);
```
Comment (LYHuang) [inline packages/app/src/components/DBRowTable.tsx:702]: This is the root cause, make input:
`JSONExtractString(Body, 'object', 'c'),JSONExtractString(Body, 'object', 'note')`
become
`["JSONExtractString(Body", "'object'", "'c')", "'note')"]`
after dedupe set(`selectColumns`).

Comment (LYHuang) [reply packages/app/src/components/DBRowTable.tsx:702]: I checked the `csv-parse`, however it does not work well with parentheses and brackets. 
Parsing option `quote` only works for same character such as: `"content"`.

Comment (wrn14897) [reply packages/app/src/components/DBRowTable.tsx:702]: got it. in this case, I think we want to make sure the unit tests cover all use cases. Does it work with the select field like `"foo,bar",field2,field3` as well?

**Comment by wrn14897** [comment]
We should also replace `splitAndTrimCSV` with new function

File: packages/app/src/components/DBRowTable.tsx (hunk 1 above)
Comment (LYHuang) [reply packages/app/src/components/DBRowTable.tsx:702]: It will split it to:
```
[
  '"foo',
  'bar"',
  'field2',
  'field3'
]
```
It should works fine if we don't get the input like: `"foo,bar","foo,differentBar",field2,field3` and also doing the dedupe.

Comment (LYHuang) [reply packages/app/src/components/DBRowTable.tsx:702]: I will work on quote stuff, will update PR later.

Comment (wrn14897) [reply packages/app/src/components/DBRowTable.tsx:702]: cool, I'd imagine the return value from the example above is `["foo,bar", "field2", "field3"]` although it's a corner case

Comment (LYHuang) [reply packages/app/src/components/DBRowTable.tsx:702]: update the split function to support single quote and double quote, feel free to check test case.

**Review by wrn14897 — APPROVED** [review]
Great fix. Thank you!

File: packages/app/src/components/DBRowTable.tsx [hunk 2]
```diff
@@ -25,11 +25,7 @@ import {
   ChartConfigWithDateRange,
   SelectList,
 } from '@hyperdx/common-utils/dist/types';
-import {
-    Granularity,
-    splitAndTrimCSV,
-    splitAndTrimWithBracket,
-} from '@hyperdx/common-utils/dist/utils';
+import { splitAndTrimWithBracket } from '@hyperdx/common-utils/dist/utils';
```
Comment (LYHuang) [inline packages/app/src/components/DBRowTable.tsx:28]: `Granularity` and `splitAndTrimCSV` is not used at here anymore, remove it since it cause conflict.

**Review by wrn14897 — APPROVED** [review]
